src/filesys.h

```diff
@@ -36,49 +36,50 @@ struct DirListNode
 	bool dir;
 };
 
+[[nodiscard]]
```

Comment:
Why do we need this? If the compiler optimizes the return value away, we should consider to disable that by CXX_FLAGS in CMake. It's also potentially error-prone because there's no such attribute for `CreateDir`, which *might* be needed to, following that logic.

Reply:
This isn't related to optimization.
It's for the compiler to warn you when you do `GetDirListing` and do not use the result. So just to reduce mistakes.

I admit it's not that important for helpers like this. The prime use case is functions like `secure_rand_fill_buf` where not handling the return value can be a security bug.